packages/number-field/test/inputs.test.ts

```diff
@@ -466,5 +466,18 @@ describe('NumberField - inputs', () => {
             expect(el.formattedValue).to.equal('10%');
             expect(el.value).to.equal(Number(0.1));
         });
+        it('does not accept non-numeric characters', async () => {
+            const el = await getElFrom(Default(Default.args));
+
+            el.focusElement.focus();
+            el.dispatchEvent(new CompositionEvent('compositionstart'));
+            await sendKeys({ type: 'あい' });
+
+            await elementUpdated(el.focusElement);
+            await nextFrame();
+
+            expect(el.focusElement.value).to.equal('100');
+            el.dispatchEvent(new CompositionEvent('compositionend'));
```

Comment:
Could we add another assertion after the `compositionend` event to ensure the value remains unchanged once composition is complete?
```suggestion
            el.dispatchEvent(new CompositionEvent('compositionend'));
                        expect(el.focusElement.value).to.equal('100');
```